src/Microsoft.DotNet.Core.Build.Tasks/LockFileCache.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Concurrent;
+using NuGet.Common;
+using NuGet.ProjectModel;
+
+namespace Microsoft.DotNet.Cli.Tasks
+{
+    internal class LockFileCache
```

Comment:
I'm not well versed in assembly load contexts.  I don't know if they provide any sort of memory barriers. 
If they don't, then the data will be cached.  If they do provide a memory barrier here, the worst case is that the lock file is parsed again.  It wouldn't affect functionality.